### PR TITLE
[Kevin Foong] iP

### PR DESCRIPTION
**Pull request by kevin9foong**
# Duke 🦑 (header)
### Kevin's CS2103T iP 👍

Features (bullet list): 
* Add Task Variations (Event, Deadline, ToDo)  
* Delete Tasks
* Search for Tasks 
* ~~GUI~~ CLI-based 💪 

Product Roadmap 🚀 (numbered list): 
1. Create core features
2. Learn new stuff 
3. Draw UML diagrams 💎 
4. **Have fun!** 

Fenced code block: 
```
system.out.println("Hi, I'm Duke at your service");
```

Tasks remaining: 
- [x] Week 4
- [x] Week 5
- [x] Week 6
- [x] _and beyond..._ 

As _Edgar W. Dijkstra_ once said ☄️ :
> Perfecting oneself is as much unlearning as it is learning. 

Hyperlink: 
[HttpStatusDogs 🐶 ](https://httpstatusdogs.com/)

This is some inline code `system.out.println("Hi, I am an inline code.");`.

### CODE REVIEW

File: src/main/java/duke/Duke.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        vbox.setAlignment(Pos.TOP_CENTER);
+        VBox.setVgrow(userOutputTextArea, Priority.ALWAYS);
+
+        Scene scene = new Scene(vbox);
+
+        primaryStage.setTitle("Duke Task Manager");
+        primaryStage.setScene(scene);
+        primaryStage.show();
+    }
+}
```
Comment (fans2619) [inline src/main/java/duke/Duke.java]: Do you want to insert a newline at the end of every java source file?

File: src/main/java/duke/tasks/Task.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.exceptions.InvalidTaskDataException;
+
+/**
+ * This class encapsulates a user-added task
```
Comment (fans2619) [inline src/main/java/duke/tasks/Task.java]: Probably missing "." at the end.

File: src/main/java/duke/tasks/Task.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            throw new InvalidTaskDataException();
+        }
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Set description of this task.
```
Comment (fans2619) [inline src/main/java/duke/tasks/Task.java]: Should it be "Sets"?

File: src/main/java/duke/tasks/Task.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    protected void setIsDone(boolean isDone) {
+        this.isDone = isDone;
+    }
+
+    private String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Convert this task's data to representation to be saved in file.
```
Comment (fans2619) [inline src/main/java/duke/tasks/Task.java]: Should it be "Converts"?

File: src/main/java/duke/tasks/TaskList.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    private void saveTasksToStorage(List<Task> tasksToSave) throws TaskFileIoException {
+        if (taskStorage == null) {
+            taskStorage = new TaskFileStorage();
+        }
+        taskStorage.saveTasks(tasksToSave);
+    }
+
+    /**
+     * Get task associated with index number in taskList.
```
Comment (fans2619) [inline src/main/java/duke/tasks/TaskList.java]: Should it be "Gets"?

**Review by fans2619 — COMMENTED** [review]
Yea it is excellent in terms of coding standard. Good job! 👍

File: src/main/java/duke/command/Command.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    /**
+     * Handles this command.
+     *
+     * @param userOutputHandler handles outputting messages to the output destination.
+     * @param taskList          handles task operations including adding, deleting, marking as done and retrieval.
+     * @throws DukeException thrown when a failure occurs within the Duke chat bot.
+     */
+    public abstract void execute(UserOutputHandler userOutputHandler, TaskList taskList)
+            throws DukeException;
```
Comment (Nature711) [inline src/main/java/duke/command/Command.java]: I feel like there's no need to put the throws part in a new line; as I've counted, putting the entire method signature in one line won't make the line longer than 120 characters :D

**Review by Nature711 — COMMENTED** [review]
LGTM. Just need to fix some minor issues in the tense used in javadoc :D

**Review by Moley456 — COMMENTED** [review]
Everything looks good! Clean and good coding in general, just a small issue with the code related to the GUI and very minor issues with JavaDocs comments that have been brought up by the other reviewers. Good job and keep it up! 👍🏻

File: src/main/java/duke/tasks/Task.java (hunk 3 above)
Comment (kevin9foong) [reply src/main/java/duke/tasks/Task.java]: Resolved. Thanks for pointing it out!

**Pull request closed without merging**